src/app/frontend/common/components/middleellipsis/middleellipsis.js

```diff
@@ -0,0 +1,82 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Based on given container and its width calculates maximum text length.
+ * Filter is used to truncate string until binary search finds optimal length to use
+ * available space.
+ *
+ * Container is outer element of 'element' and is used as bounding box to allow stretching of
+ * inner element and not exceed boundaries of outer container. It's the best to style container
+ * with display: [block|inner-block] and width: 100% and disable text wrapping for the element.
```

Comment:
Please elaborate that a bit because I'm not sure what exactly should I change here :)